### DISCUSSION TITLE
Form is invalid but errors field is empty because of error validation not triggered on first mount

### DISCUSSION THREAD

**Discussion by bryanprimus** [react-hook-form/react-hook-form#9947] [Q&A]
Sandbox:

https://stackblitz.com/edit/vitejs-vite-gkpsxn?file=src/App.tsx

I have seen this discussion and seems like it's the behavior of RHF itself to prevent rerender

Then what would be the best approach to trigger form validation so that the the errors field is not empty while the form is not valid?

on my sandbox, i use `trigger` on useEffect. but i wonder if there is a better approach,

I also see this approach https://github.com/react-hook-form/react-hook-form/discussions/4079#discussioncomment-329688, is this also a way to go?

**Comment by bluebill1049** [comment]
react hook form has a mode for validation strategy, which means we optimise the error state on your behalf for consistent UX if you want to manage the error state on your own. Yes, your approach is fine.
